### DISCUSSION TITLE
Problem with custom backend

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/vtm#829] [General]
Hello,

I am trying to implement a custom backend with VTM. It works but sometimes all elements vanish as shown here: https://youtu.be/mbkrTPZiho4

I know that this is probably out of scope but do you still have an idea what could cause the problem?

**Comment by demiantres** [comment]
After some research I found the problem: The VTM library does not always properly enables vertex arrays by calling "enableVertexAttribArray()" before "vertexAttribPointer()" (for instance in the BitmapBucket and LineBucket, but also other buckets).

**Reply by devemux86** [reply]
We cannot test third-party backends or know if they are implemented correctly.

The most important thing is if VTM works properly in its supported modules.

Anyway there is `GLState.enableVertexArrays` in many locations and in buckets.

**Reply by demiantres** [reply]
My initial suspicion was that the reason is the custom backend but it turned out that the enableVertexAttribArray method is not correctly called. You can test this by manually disabling all vertex arrays before each rendering loop and suddenly (some) tiles or polygon shapes will not render correctly (for instance raster tiles).

**Comment by devemux86** [comment]
Reports must be reproduced in repository samples using the official code.
